wdl-grammar/src/v1/lint/double_quotes.rs

```diff
@@ -0,0 +1,148 @@
+//! All quotes should be double quotes.
+
+use std::collections::VecDeque;
+
+use nonempty::NonEmpty;
+use pest::iterators::Pair;
+use wdl_core::concern::code;
+use wdl_core::concern::lint;
+use wdl_core::concern::lint::Rule;
+use wdl_core::concern::lint::TagSet;
+use wdl_core::concern::Code;
+use wdl_core::file::Location;
+use wdl_core::Version;
+
+use crate::v1;
+
+///
+#[derive(Debug)]
+pub struct DoubleQuotes;
+
+impl<'a> DoubleQuotes {
+    /// Creates a warning for strings defined using single-quotes
+    fn single_quote_string(&self, location: Location) -> lint::Warning
+    where
+        Self: Rule<&'a Pair<'a, v1::Rule>>,
+    {
+        // SAFETY: this error is written so that it will always unwrap.
+        lint::warning::Builder::default()
+            .code(self.code())
+            .level(lint::Level::Low)
+            .tags(self.tags())
+            .push_location(location)
+            .subject("string defined with single quotes")
+            .body("All strings should be defined using double quotes.")
+            .fix("Change the single quotes to double quotes.")
+            .try_build()
+            .unwrap()
+    }
+}
+
+impl<'a> Rule<&'a Pair<'a, v1::Rule>> for DoubleQuotes {
+    fn code(&self) -> Code {
+        // SAFETY: this manually crafted to unwrap successfully every time.
+        Code::try_new(code::Kind::Warning, Version::V1, 12).unwrap()
+    }
+
+    fn tags(&self) -> lint::TagSet {
+        TagSet::new(&[lint::Tag::Naming])
```

Comment:
I'm not sure this falls ender `Naming`? It's definitely `Style` (which is being auto-added bc of the `Naming` tag), and I think this rule calls for a new tag called `Consistency`. And if we're adding a `Consistency` tag, that should be added to a few of the existing rules: `whitespace`, `no_curly_commands`, `mixed_indentation`, `snake_case`? Open to hearing dissenting opinions on any of these specifics here.

Reply:
It is labeled `Naming` in the discussion (https://github.com/stjude-rust-labs/wdl/discussions/11), but I agree. I'm not sure that `Naming` is correct.

Reply:
`mixed_indentation` uses `Style|Spacing|Clarity`. Is `Clarity` sufficient here or does `Consistency` convert something different?

Reply:
`snake_case` similarly uses `Naming|Style|Clarity`

Reply:
Hm. They are similar, but different.

Generally, I'd argue "too many tags" is a better problem to have than "not enough tags". Tags serve the (primary?) function of making the docs easier to navigate and the (secondary?) function of providing easy shorthand for groups of rules that a user would want to toggle at once. (For the sake of argument, we will pretend those functions exist today lol)

If those are our two main goals for tags, I don't think there's a real problem with having too many. The biggest downside (I can see) is that having lots of tags muddies the warning's shorthand notation (`[v1::W004::[Spacing, Style, Clarity]::High]`).

Reply:
What is the final decision here? Only `Clarity` or `Clarity|Style`?

Reply:
I misread @adthrasher 's comments. Let me try addressing this again lol (so @simojoe hold on before making any changes)

>  I would tend to favor `Clarity` here, ironically, for consistency with the already implemented rules.

I don't see how `Clarity` applies to `double_quotes`. This rule isn't a question of "what's clearer?".

As authors, everyone is faced with what is a more or less identical choice between single and double quotes. We've arbitrarily chosen double quotes to be the standard because switching back and forth is bad. But as long as it's consistent, it doesn't matter.

That's why I want to add `Consistency` here. And then in hindsight, `Consistency` _could_ apply to some of our other rules.

What's the argument against having both `Consistency` and `Clarity`?

Reply:
@a-frantz I think it's worth trying to avoid too many tags if we can. When I look at the rules `snake_case`, `no_curly_commands`, and `mixed_indentation`, in all three cases we are choosing a preferred style for the purpose of `Clarity`. Technically, all would be `Consistency` as well.

Reply:
options:
1. no new `Consistency` tag
    * IMO `Clarity` doesn't quite fit for `double_quotes`. Which means `double_quotes` would only have the tag `Style`.
2. replace all instances of `Clarity` with `Consistency`
    * `double_quotes` gets both `Consistency` and `Style`
3. Have both `Clarity` and `Consistency`.
    * They appear together for `snake_case`, `no_curly_commands`, and `mixed_indentation` _but not_ `double_quotes` (which only gets `Consistency`)
 4. ??? is there anything else worth considering?
 
I would rule out number 2. Sounds like you want to rule out number 3. I don't feel strongly about number 1, so maybe that's the compromise we settle on?

Reply:
Copying/summarizing conversation with @a-frantz from Slack, but the conclusion is that `double_quotes` should get `Style` and `Clarity` tags.

Personally, I view `consistency` as a necessary component of `clarity`. If something is inconsistent (e.g. mixing single and double quotes), it lacks clarity. The choice of double quotes vs single quotes comes down to a matter of `Style` in my opinion as well. So `Style|Clarity` seems to fit.

There will also likely be an effort in the future to define a set of tags to support and write descriptions for each. With that in mind, keeping a smaller set of tags now makes that easier. We can always add tags to existing rules as part of that effort.